core-tests/src/test/java/org/mule/runtime/core/internal/registry/MuleRegistryHelperTransformerLookupTestCase.java

```diff
@@ -31,15 +34,20 @@
 public class MuleRegistryHelperTransformerLookupTestCase extends AbstractMuleTestCase {
 
   private static final DataType ORANGE_DATA_TYPE = DataType.fromType(Orange.class);
+  private static final DataType APPLE_DATA_TYPE = DataType.fromType(Apple.class);
 
   private final Registry registry = mock(Registry.class);
   private final MuleContext muleContext = mock(MuleContext.class);
   private final MuleRegistryHelper muleRegistryHelper = new MuleRegistryHelper(registry, muleContext);
   private final Converter stringToOrange = new MockConverterBuilder().from(DataType.STRING).to(ORANGE_DATA_TYPE).build();
   private final Converter orangeToString = new MockConverterBuilder().from(ORANGE_DATA_TYPE).to(DataType.STRING).build();
+  private final Converter appleToString = new MockConverterBuilder().from(APPLE_DATA_TYPE).to(DataType.STRING).build();
 
   @Before
   public void setUp() throws Exception {
+    Collection transformers = Collections.singletonList(appleToString);
+    when(registry.lookupObjects(Transformer.class)).thenReturn(transformers);
```

Comment:
this is not how transformers are resolved. They are registered by a post processor on initialisation time

Reply:
this is not a good solution for the root problem. The root problem is that TransformerResolvers from the domain are not being considered by the app. This is a hack that will a) severely hurt performance when running in a domain b) not take advantage of any fix/improvements done in the Transformer resolution framework